tmf/org.eclipse.tracecompass.tmf.core/src/org/eclipse/tracecompass/tmf/core/dataprovider/IDataProviderDescriptor.java

```diff
@@ -61,6 +61,11 @@ public enum ProviderType {
          * @since 6.1
          */
         DATA_TREE,
+        /**
+         * A provider for bar chart.
+         * @since 10.2
+         */
+        BAR_CHART,
```

Comment:
I think the new provider type should be about new xy provider where the x-axis is not the trace time. The chart type is part of the `DisplayType`. We could have other xy-charts like line, scatter etc. with a time-less x-axis.

Reply:
I saw you settle in GENERIC_XY which fits.